### PR TITLE
refactor(internal-icons): rework building

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

Moves the other "internal" icons to the internal folder.

Builds all icons the same way.

Removes svgr rollup and webpack.

### CODE REVIEW

File: .storybook/webpack.config.js [hunk 1]
```diff
@@ -56,41 +56,6 @@ module.exports = ({ config }) => {
         },
       ],
     },
-    // For svg icons, we want to get them transformed into React components
-    // when we import them.
-    {
```
Comment (montezume) [inline .storybook/webpack.config.js]: now we build all icons the same way, no need for this.

File: package.json [hunk 2]
```diff
@@ -119,10 +119,7 @@
     "@storybook/theming": "5.2.6",
     "@svgr/core": "4.3.3",
     "@svgr/plugin-jsx": "4.3.3",
-    "@svgr/plugin-prettier": "4.3.2",
```
Comment (montezume) [inline package.json]: this wasn't used

File: scripts/generate-icon-exports.js [hunk 3]
```diff
@@ -9,101 +9,161 @@ const { stripIndents } = require('common-tags');
...
+  },
+  {
+    indexPath: internalIndexPath,
+    svgPath: internalIconsPath,
+    rawExportsPath: internalRawExportsPath,
+    iconComponentsPath: internalIconComponentsPath,
+  },
+];
+
+iconsToProcess.forEach(data => {
```
Comment (montezume) [inline scripts/generate-icon-exports.js]: now we go through two "rounds".

first time for the public icons, second for the internal.

File: src/components/inputs/radio-input/radio-option.styles.js [hunk 4]
```diff
@@ -5,6 +5,12 @@ const getContainerStyles = props => {
   const baseStyles = css`
     display: flex;
     align-items: center;
+
+    /* reset from createStyledIcon */
```
Comment (montezume) [inline src/components/inputs/radio-input/radio-option.styles.js]: just overrides the default styles added by our icons.

File: scripts/generate-icon-exports.js [hunk 5]
```diff
@@ -9,101 +9,161 @@ const { stripIndents } = require('common-tags');
...
+        Component.displayName = '${displayName}';
+
+        // we do this to enable treeshaking
+        // please see https://github.com/alex996/react-css-spinners/issues/1
+        if (process.env.NODE_ENV !== "production") {
+          Component.propTypes = iconPropTypes;
+        }
+
+        export default Component;
+      `;
```
Comment (tdeekens) [inline scripts/generate-icon-exports.js]: Starting to wonder if we could some time dump them into a `icon.template.js` or seomthing.

Comment (montezume) [reply scripts/generate-icon-exports.js]: people: redux is too much boilerplate! everything is in different files.
also people: hey, we have 3 things in one file. let's put it in three.

😄

Comment (tdeekens) [reply scripts/generate-icon-exports.js]: I see your point but was never the one who hated Redux for boilerplate :)

**Comment by emmenko** [comment]
:100:

**Pull request merged**